testgrid/cmd/entomologist/main.go

```diff
@@ -0,0 +1,212 @@
+	var o options
+
+	if err := o.parseArgs(flag.CommandLine, os.Args[1:]); err != nil {
+		logrus.WithError(err).Fatal("Invalid Flags")
+	}
+
+	return o
+}
+
+func main() {
+	opt := parseOptions()
+
+	getToken := func() []byte {
+		return opt.ghCredentials
+	}
+
+	ghct := github.NewClient(getToken, "", opt.ghProxyEndpoint)
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+
+	client, _ := io.NewOpener(ctx, opt.gcsCredentials)
+	if client == nil {
+		logrus.Fatalf("Empty Credentials (at %s) not allowed for write operation", opt.gcsCredentials)
+	}
+
+	gcsFile := fmt.Sprintf("/bugs-%s-%s", opt.organization, opt.repository)
+	writer, _ := client.Writer(ctx, opt.gcsPath+gcsFile)
+	defer writer.Close()
+
+	// kick off goroutines
+	poll := func() {
+		issueState, err := pinIssues(ghct, opt)
+		if err != nil {
+			logrus.Fatalf("Could not get issues: %e", err)
+		}
+
+		out, _ := proto.Marshal(issueState)
+		n, _ := writer.Write(out)
+
+		logrus.Infof("Sending %d characters to GCS", n)
+	}
+
+	if opt.oneshot {
+		poll()
+		return
+	}
+
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	wait.Until(poll, defaultPollTime, stopCh)
+
+	sigTerm := make(chan os.Signal, 1)
+	signal.Notify(sigTerm, syscall.SIGTERM)
+	signal.Notify(sigTerm, syscall.SIGINT)
+	<-sigTerm
+
+	logrus.Info("Entomologist is closing...")
+}
+
+var tre = regexp.MustCompile(`(?m)^target:(.+)$`)
+
+func getIssues(client githubClient, o options) ([]github.Issue, error) {
+	issuesAndPRs, err := client.ListIssues(o.organization, o.repository)
+
+	issues := issuesAndPRs[:0]
+	for _, issue := range issuesAndPRs {
+		if !issue.IsPullRequest() {
+			issues = append(issues, issue)
+		}
+	}
+
+	return issues, err
+}
+
+// Algorithm for detecting and formatting test targets in issues
+func pinIssues(client githubClient, o options) (*issue_state.IssueState, error) {
+	issues, err := getIssues(client, o)
+
+	logrus.Infof("Found %d Issues in %s/%s", len(issues), o.organization, o.repository)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var results []*issue_state.IssueInfo
+
+	for _, issue := range issues {
+		newResult := issue_state.IssueInfo{
+			IssueId: strconv.FormatInt(int64(issue.Number), 10),
+			Title:   issue.Title,
+			RowIds:  make([]string, 0),
+		}
+
+		matches := tre.FindAllStringSubmatch(issue.Body, -1)
+		for _, match := range matches {
+			newResult.RowIds = append(newResult.RowIds, strings.TrimSpace(match[1]))
+		}
+
+		// todo(chases2): separate the comments API calls into their own goroutines
+		comments, err := client.ListIssueComments(o.organization, o.repository, issue.Number)
+		if err != nil {
+			return nil, err
+		}
+		for _, comment := range comments {
+			matches := tre.FindAllStringSubmatch(comment.Body, -1)
+			for _, match := range matches {
+				newResult.RowIds = append(newResult.RowIds, strings.TrimSpace(match[1]))
+			}
+		}
+
+		if len(newResult.RowIds) != 0 {
+			results = append(results, &newResult)
```

Comment:
consider not building `newResult` until you need it down here; it looks like most of the logic above is about creating an array for `newResult.RowIds`

Reply:
I have done so, and renamed the slice of targets this routine is building to "targets"